### PR TITLE
feat(tile, tile-group): add components

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

https://jsw.ibm.com/browse/ADCMS-6691

### Description

Migrates the `tile` and `tile-group` components from `carbon-for-aem` into `carbon-for-ibmdotcom`

### Changelog

**New**

- adds `c4d-tile`
- adds `c4d-tile-group`

### CODE REVIEW

**Review by areagan1030 — APPROVED** [review]

File: packages/web-components/src/components/tile/__stories__/tile-group.stories.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html } from 'lit-element';
```
Comment (m4olivei) [inline packages/web-components/src/components/tile/__stories__/tile-group.stories.ts:10]: We should take the opportunity here to update lit imports to the recommended form. See the [v2 upgrade guide](https://lit.dev/docs/v2/releases/upgrade/#update-packages-and-import-paths). I had a [PR open to do this on CAEM](https://github.ibm.com/webstandards/carbon-for-aem/pull/196/files), never got around to tidying it up, but it stands as a good guide to the changes we need. Doing this change will align well with existing components in the repo.

File: packages/web-components/src/components/tile/__stories__/tile.stories.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  </c4d-tag-group>
+`;
+
+const pictogramContent = html`
+  <svg
+    slot="pictogram"
+    focusable="false"
+    preserveAspectRatio="xMidYMid meet"
+    xmlns="http://www.w3.org/2000/svg"
+    data-autoid="dds--card__pictogram"
```
Comment (m4olivei) [inline packages/web-components/src/components/tile/__stories__/tile.stories.ts:32]: `dds` prefix here. We can probably just drop this `data` attribute I'd think? Was probably copy 🍝  from somewhere long ago.

File: packages/web-components/src/components/tile/tile.scss [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+@use '@carbon/grid' as *;
+@use '@carbon/ibmdotcom-styles/scss/globals/utils/flex-grid' as *;
+
+// @use '@carbon/styles/scss/type' as *;
+// @use '@carbon/styles/scss/theme' as *;
+// @use '@carbon/styles/scss/utilities';
+// @use '@carbon/styles/scss/spacing' as *;
+// @use '@carbon/styles/scss/reset' as *;
+// @use '@carbon/grid' as *;
+// @use '@carbon/ibmdotcom-styles/scss/globals/utils/flex-grid' as *;
```
Comment (m4olivei) [inline packages/web-components/src/components/tile/tile.scss:25]: Remove commented out code.

File: packages/web-components/src/components/tile/tile.scss [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+      display: grid;
+      grid-template-columns: 100%;
+      grid-template-rows: 1fr;
+      height: 100%;
+
+      &::before {
+        content: '';
+        display: block;
+        /* stylelint-disable-next-line property-no-unknown */
+        aspect-ratio: var(--caem-tile-aspect-ratio, 16 / 9);
```
Comment (m4olivei) [inline packages/web-components/src/components/tile/tile.scss:60]: Any reason to keep the `caem` prefix here, or can we update this to a `c4d` prefix?

File: packages/web-components/src/components/tile/tile.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+ * @license
+ *
+ * Copyright IBM Corp. 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { LitElement, html, property, state, query } from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map.js';
```
Comment (m4olivei) [inline packages/web-components/src/components/tile/tile.ts:11]: Any directive imports also need updating to [use the `lit` package instead](https://lit.dev/docs/v2/releases/upgrade/#update-directive-imports).

**Review by m4olivei — CHANGES_REQUESTED** [review]
Noticed a couple of things from a static review perspective. Also, review CI jobs, looks like [at least stylelint is angry](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/actions/runs/11638687608/job/32413835196?pr=12091).

Also comparing the componets on CAEM vs here:

The image ratios are different. Not sure which is correct:

image

The Tile Group Default and With Image stories lost their knobs configuration.

image

File: packages/web-components/src/components/tile/__stories__/tile-group.stories.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+import { boolean, select, text } from '@storybook/addon-knobs';
+
+const ctaTypeOptions = Object.values(CTA_TYPE).filter((value) => !!value);
+
+export default {
+  title: 'Components/Tile Group',
+  parameters: {
+    ...readme.parameters,
+    hasStoryPadding: true,
+    knobs: () => {
```
Comment (m4olivei) [inline packages/web-components/src/components/tile/__stories__/tile-group.stories.ts:28]: The knobs are still not showing up for the tile stories. Looking at other components, it looks like this property is expected to be an object keyed by the component, with a value of this function, eg.

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/quote/__stories__/quote.stories.ts#L64-L88

**Review by m4olivei — CHANGES_REQUESTED** [review]

**Review by m4olivei — APPROVED** [review]
Looking good now!

**Pull request merged**